src/main/java/org/terasology/scenario/internal/ui/LogicTreeMenuTreeBuilder.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.scenario.internal.ui;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.contextMenu.MenuTree;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class LogicTreeMenuTreeBuilder {

```

Comment:
Also using strings as keys to get something is not so ideal. Especially as they are actually texts that might later get translated. If you want to go for constants you could go for an enum although  you problably
just get a nice syntax by just passing method references as arguments  to addOption